### ISSUE TITLE
QPDF as a reporting tool

### ISSUE THREAD

**Issue by datalogics-mdawson** [qpdf/qpdf#690]
I read in your description that QPDF provides a means for developers to "look at the innards of a PDF file to learn more about how they work." I'm wondering if the software offers a capacity to inspect a PDF document and provide detailed information about the nature of that file, such as the file size and number of pages, PDF/A compliance, compression details, metadata, and whether the file is damaged or corrupted, and whether the file contains JavaScript, embedded fonts, annotations, and the like. How would this data be reported?

**Comment by jberkenbilt** [comment]
qpdf doesn't offer the ability to answer those specific questions out of the box, other than number of pages (`qpdf --show-npages file.pdf`). Rather it makes it possible to use the library or command line to probe into the structure of the PDF file to determine those kinds of things if you understand the PDF spec well enough to know how to answer those questions. It handles all the PDF file parsing, decompression, etc., but not the higher level details in general. There are a few higher-level features that qpdf has, such as listing attachments. qpdf can test whether the file is damaged or corrupted to a certain extent and is pretty good about structural damage, but it does very little checking on semantics and almost no checking on page contents.

In other words, it's a pretty low-level tool. If one wanted to write something to answer the kinds of questions you are looking for, qpdf would be a good choice as the PDF library to use for that purpose, but it doesn't already have the answers. Hope that helps.
